treeDetectLmax.py

```diff
@@ -0,0 +1,276 @@
+        geometry.Destroy()
+        feature.Destroy()
+    # Cleanup
+    shapeData.Destroy()
+    # Return
+    return shapePath
+
+
+def kmlsave(kmlPath, trees):
+    ''' Save data to a kml file
+        @param  kmlPath       Input path to the kml which will be created (Str)
+        @param  trees         Input tree data (dictionnary)
+    '''
+    geoLocations = np.column_stack((trees['xpos'],trees['ypos']))
+    height = trees['height']
+    
+    driver = ogr.GetDriverByName('KML')
+
+    # Create shapeData
+    kmlPath = validateKmlPath(kmlPath)
+    if os.path.exists(kmlPath):
+        os.remove(kmlPath)
+        
+    kmlData = driver.CreateDataSource(kmlPath)
+    # Create spatialReference
+    outSpatialRef = osr.SpatialReference()
+    outSpatialRef.ImportFromEPSG(4326) # output SpatialReference
+    
+    spatialReference = getSpatialReferenceFromWkt(trees['prj_wkt'])
+    transform = osr.CoordinateTransformation(spatialReference, outSpatialRef)
+
+    # Create layer
+    layerName = os.path.splitext(os.path.split(kmlPath)[1])[0]
+    layer = kmlData.CreateLayer(layerName, outSpatialRef, ogr.wkbPoint)
+    
+    field_height = ogr.FieldDefn("H", ogr.OFTReal)
+    field_height.SetWidth(4)
+    field_height.SetPrecision(2)
+    
+    layer.CreateField(field_height)
+    
+    for pointIndex, geoLocation in enumerate(geoLocations):
+        # Create point
+        geometry = ogr.Geometry(ogr.wkbPoint)
+        geometry.SetPoint(0, geoLocation[0], geoLocation[1])
+        geometry.Transform(transform)
+        
+        # Create feature
+        feature = ogr.Feature(layer.GetLayerDefn())
+        feature.SetGeometry(geometry)
+        feature.SetFID(pointIndex)
+        feature.SetField("H",float(height[pointIndex]))
+        
+        # Save feature
+        layer.CreateFeature(feature)
+        # Cleanup
+        geometry.Destroy()
+        feature.Destroy()
+    # Cleanup
+    kmlData.Destroy()
+    # Return
+    return kmlPath
+    
+
+def getSpatialReferenceFromWkt(Wkt):
+    '''Return GDAL spatial reference object from proj4 string'''
+    spatialReference = osr.SpatialReference()
+    spatialReference.ImportFromWkt(Wkt)
+    return spatialReference
+    
+def validateShapePath(shapePath):
+    '''Validate shapefile extension'''
+    return os.path.splitext(str(shapePath))[0] + '.shp'
+    
+def validateKmlPath(kmlPath):
+    '''Validate kml extension'''
+    return os.path.splitext(str(kmlPath))[0] + '.kml'    
+    
+# convert pixel coordinates to geospatial coordinates
+def ApplyGeoTransform(inx,iny,gt):
+    ''' Apply a geotransform
+        @param  inx       Input x coordinate (double)
+        @param  iny       Input y coordinate (double)
+        @param  gt        Input geotransform (six doubles)
+        @return outx,outy Output coordinates (two doubles)
+    '''
+    outx = gt[0] + inx*gt[1] + iny*gt[2]
+    outy = gt[3] + inx*gt[4] + iny*gt[5]
+    return (outx,outy)    
+
+
+# if __name__ == "__main__":
+    
```

Comment:
What are you calling "from a script" ? Something outside of QGIS to test the functions ?

If it is that you want to do, as I understand, it would be possible be creating a virtualenv containing all the dependencies which are needed... But I am not sure that I completely understood your question...

Could you expose a bit more what you exactly want to do ?